apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/PgNumericTests.cs

```diff
@@ -0,0 +1,325 @@
+
+        [Theory]
+        [InlineData("NaN", "NaN")]
+        [InlineData("Nan", "nan")]
+        [InlineData("0.00", "000")]
+        [InlineData("-1.0", "-1.000")]
+        [InlineData("-1", "-1.0")]
+        [InlineData("1", "1.0")]
+        [InlineData("0.00000009", "0.00000009")]
+        [InlineData("-0.00", "0.0")]
+        public void CompareToEqualValueTest(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.CompareTo(number2);
+            Assert.Equal(0, result);
+        }
+
+        [Theory]
+        [InlineData("9999999999999", "NaN")]
+        [InlineData("0.00", "001")]
+        [InlineData("-1.0", "1.000")]
+        [InlineData("-10.0", "0.00000009000000000")]
+        [InlineData("0", "1.00000000")]
+        [InlineData("1", "1.01")]
+        [InlineData("0.00000009", "0.0000008")]
+        public void CompareToGreaterValueTest(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.CompareTo(number2);
+            Assert.Equal(-1, result);
+        }
+
+        [Theory]
+        [InlineData("NaN", "99999999999999999999999999999")]
+        [InlineData("99.99", "99.90")]
+        [InlineData("1.0", "-1.000")]
+        [InlineData("-1", "-55.0")]
+        [InlineData("0", "-0.0009")]
+        [InlineData("0.00009", "0.00000009")]
+        public void CompareToSmallerValueTest(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.CompareTo(number2);
+            Assert.Equal(1, result);
+        }
+
+        [Theory]
+        [InlineData("1.00", "1")]
+        [InlineData("1", "1.00")]
+        [InlineData("NaN", "NaN")]
+        [InlineData("NAN", "Nan")]
+        [InlineData("1.0000", "1.000")]
+        [InlineData("1.00", "1.0")]
+        [InlineData("-1.00", "-1.0")]
+        [InlineData("-0.00", "0.0")]
+        public void EqualsTest(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.Equals(number2);
+            Assert.Equal(true, result);
+        }
+
+        [Theory]
+        [InlineData("1.00", "1")]
+        [InlineData("1", "1.00")]
+        [InlineData("NaN", "NaN")]
+        [InlineData("1.0000", "1.000")]
+        [InlineData("999999999999999999999999999999999999999999999999.999999999999999999999999999999999999999", "999999999999999999999999999999999999999999999999.999999999999999999999999999999999999999")]
+        [InlineData("-999999999999999999999999999999999999999999999999.999999999999999999999999999999999999999", "-999999999999999999999999999999999999999999999999.999999999999999999999999999999999999999")]
```

Comment:
These would benefit from instead of using nines, using 12345678912345.... it's easier to count the number blocks than a lot of nines.